### PR TITLE
Create FederationFactoryTest

### PR DESCRIPTION
**Pull request by julia-zack**
Test FederationFactory works as expected

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            bridgeConstants = BridgeTestNetConstants.getInstance();
+            networkParameters = bridgeConstants.getBtcParams();
+            federation = createP2shErpFederation();
+            version = federation.getFormatVersion();
+            assertEquals(P2SH_ERP_FEDERATION.getFormatVersion(), version);
+        }
+
+    }
+
+    public Federation createStandardMultisigFederation() {
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java:178]: These 3 methods should probably be private, right?

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+class FederationFactoryTest {
+    private BridgeConstants bridgeConstants;
+    private NetworkParameters networkParameters;
+    private List<FederationMember> federationMembers;
+    private List<BtcECKey> defaultKeys;
+    private Instant creationTime;
+    private long creationBlockNumber;
+    private int defaultThreshold;
+    private List<BtcECKey> emergencyKeys;
+    private int emergencyThreshold;
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java:36]: I think these thresholds values are not used anywhere

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java (hunk 1 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java:178]: done

**Review by marcos-iov — APPROVED** [review]

**Review by jeremy-then — APPROVED** [review]
LGTM.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            assertNotEquals(federationPostRSKIP201PreRSKIP284, federationPostRSKIP201PostRSKIP284PreRSKIP293);
+
+            // build non standard fed
+            when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+            ErpFederation federationPostRSKIP201PostRSKIP284PostRSKIP293 = createNonStandardErpFederation();
+            version = federationPostRSKIP201PostRSKIP284PostRSKIP293.getFormatVersion();
+
+            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);
+            assertNotEquals(federationPostRSKIP201PreRSKIP284, federationPostRSKIP201PostRSKIP284PostRSKIP293);
+            assertNotEquals(federationPostRSKIP201PostRSKIP284PreRSKIP293, federationPostRSKIP201PostRSKIP284PostRSKIP293);
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java:122]:
```suggestion
            when(activations.isActive(ConsensusRule.RSKIP201)).thenReturn(true);
            ErpFederation federationPostRSKIP201 = createNonStandardErpFederation();
            version = federationPostRSKIP201.getFormatVersion();

            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);

            // build non standard fed with csv unsigned big endian
            when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
            ErpFederation federationPostRSKIP284 = createNonStandardErpFederation();
            version = federationPostRSKIP284.getFormatVersion();

            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);
            assertNotEquals(federationPostRSKIP201, federationPostRSKIP284);

            // build non standard fed
            when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
            ErpFederation federationPostRSKIP293 = createNonStandardErpFederation();
            version = federationPostRSKIP293.getFormatVersion();

            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);
            assertNotEquals(federationPostRSKIP201, federationPostRSKIP293);
            assertNotEquals(federationPostRSKIP284, federationPostRSKIP293);
```

What do you think about renaming those variables in this way? We can assume that for an N rskip, lower rskip are active, and vice versa.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+            assertEquals(federationPostRSKIP201PreRSKIP284, federationPostRSKIP201PostRSKIP284PreRSKIP293);
+
+            // build non standard fed
+            when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+            ErpFederation federationPostRSKIP201PostRSKIP284PostRSKIP293 = createNonStandardErpFederation();
+            version = federationPostRSKIP201PostRSKIP284PostRSKIP293.getFormatVersion();
+
+            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);
+            assertNotEquals(federationPostRSKIP201PreRSKIP284, federationPostRSKIP201PostRSKIP284PostRSKIP293);
+            assertNotEquals(federationPostRSKIP201PostRSKIP284PreRSKIP293, federationPostRSKIP201PostRSKIP284PostRSKIP293);
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java:151]: Same here

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java (hunk 3 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java:122]: good idea, is clearer

Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java]: Better now

**Review by nathanieliov — COMMENTED** [review]
LGTM

**Review by nathanieliov — APPROVED** [review]

**Pull request merged**